clients/src/main/java/org/apache/kafka/clients/consumer/internals/RequestFuture.java

```diff
@@ -100,11 +100,10 @@ public RuntimeException exception() {
      * @param value corresponding value (or null if there is none)
      */
     public void complete(T value) {
-        if (isDone)
+        if (!isDone.compareAndSet(false, true))
```

Comment:
It shouldn't be possible, but yeah, it's nice to have the protection in the code for the future. I had some buggy logic even while testing this patch which caused a future to be completed more than once.